File: src/main/java/it/gabrieletondi/telldontaskkata/domain/Order.java
package it.gabrieletondi.telldontaskkata.domain;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import static it.gabrieletondi.telldontaskkata.domain.OrderStatus.CREATED;

@NoArgsConstructor
public class Order {
    public Order(BigDecimal total, String currency, List<OrderItem> items, BigDecimal tax, OrderStatus status) {
        this.total = total;
        this.currency = currency;
        this.items = items;
        this.tax = tax;
        this.status = status;
    }

    private BigDecimal total;
    private String currency;
    private List<OrderItem> items=new ArrayList<>();
    private BigDecimal tax;
    private OrderStatus status;
    private int id;

    public BigDecimal getTotal() {
        return total;
    }


    public void setTotal(BigDecimal total) {
        this.total = total;
    }
    public void addTotal(BigDecimal amount){
        setTotal(this.total.add(amount));
    }

    public void addTax(BigDecimal tax){
        setTax(this.tax.add(tax));
    }
    public String getCurrency() {
        return currency;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public List<OrderItem> getItems() {
        return items;
    }

    public void setItems(List<OrderItem> items) {
        this.items = items;
    }

    public BigDecimal getTax() {
        return tax;
    }

    public void setTax(BigDecimal tax) {
        this.tax = tax;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void addOrderItem(OrderItem orderItem){
        items.add(orderItem);
    }

    public Boolean isOrderStatus(OrderStatus orderStatus){
        return this.getStatus().equals(orderStatus);
    }
}

File: typescript/src/domain/Order.ts
import OrderItem from './OrderItem';
import { OrderStatus } from './OrderStatus';

class Order {
  private total: number;
  private currency: string;
  private items: OrderItem[];
  private tax: number;
  private status: OrderStatus;
  private id: number;

  public getTotal(): number {
      return this.total;
  }

  public setTotal(total: number): void  {
      this.total = total;
  }

  public getCurrency(): string {
      return this.currency;
  }

  public setCurrency(currency: string): void {
      this.currency = currency;
  }

  public getItems(): OrderItem[] {
      return this.items;
  }

  public setItems(items: OrderItem[]): void {
      this.items = items;
  }

  public getTax(): number {
      return this.tax;
  }

  public setTax(tax: number): void {
      this.tax = tax;
  }

  public getStatus(): OrderStatus {
      return this.status;
  }

  public setStatus(status: OrderStatus): void {
      this.status = status;
  }

  public getId(): number {
      return this.id;
  }

  public setId(id: number): void {
      this.id = id;
  }
}

export default Order;

